hack/maistra_v1_servicemeshcontrolplane_cr_basic.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: maistra.io/v1
+kind: ServiceMeshControlPlane
+metadata:
+  name: basic-install
+spec:
+  # NOTE, if you remove all children from an element, you should remove the
+  # element too.  An empty element is interpreted as null and will override all
+  # default values (i.e. no values will be specified for that element, not even
+  # the defaults baked into the chart values.yaml).
+  istio:
+    global:
+      proxy:
+        # constrain resources for use in smaller environments
+        resources:
+          requests:
+            cpu: 100m
+            memory: 128Mi
+          limits:
+            cpu: 500m
+            memory: 128Mi
+
+    gateways:
+      istio-egressgateway:
+        # disable autoscaling for use in smaller environments
+        autoscaleEnabled: false
+      istio-ingressgateway:
+        # disable autoscaling for use in smaller environments
+        autoscaleEnabled: false
+        # set to true to enable IOR
+        ior_enabled: false
+
+    mixer:
+      policy:
+        # disable autoscaling for use in smaller environments
+        autoscaleEnabled: false
+
+      telemetry:
+        # disable autoscaling for use in smaller environments
+        autoscaleEnabled: false
+        # constrain resources for use in smaller environments
+        resources:
+          requests:
+            cpu: 100m
+            memory: 1G
+          limits:
+            cpu: 500m
+            memory: 4G
+
+    pilot:
+      # disable autoscaling for use in smaller environments
+      autoscaleEnabled: false
+      # increase random sampling rate for development/testing
+      traceSampling: 100.0
+
+    kiali:
+      ## Changed on kiali/hack folder to false
+      ## Kiali will be installed using cluster-openshift.sh steps not via the istio-operator
+      # change to false to disable kiali
+      enabled: false
+    
+      # to use oauth, remove the following 'dashboard' section (note, oauth is broken on OCP 4.0 with kiali 0.16.2)
+      # create a secret for accessing kiali dashboard with the following credentials
+      dashboard:
+        user: admin
+        passphrase: admin
+
+    tracing:
+      # change to false to disable tracing (i.e. jaeger)
```

Comment:
If users change this to false, the cluster-openshift.sh script will fail to wait because the list of expected apps include all the jaeger pods (i.e. see the changes to the script where "jaeger-collector jaeger-query" are included in "expected_apps". So the while loop will be waiting for jaeger that will never appear.

We already have a "-nj" (--no-jaeger) command line option. We should enhance it. Right now that just causes the script to remove jaeger AFTER it has been installed. We can change it to turn off this tracing.enabled flag AND change the expected_apps:

So in the line that sets expected_apps we change it to be:'

```
if [ "${REMOVE_JAEGER} == "true" ]; then
      expected_apps=(grafana istio-citadel istio-egressgateway istio-galley istio-ingressgateway istio-pilot istio-policy istio-sidecar-injector istio-telemetry prometheus)
else
      expected_apps=(grafana istio-citadel istio-egressgateway istio-galley istio-ingressgateway istio-pilot istio-policy istio-sidecar-injector istio-telemetry jaeger-collector jaeger-query prometheus)
fi
```

Secondly, because the user doesn't want jaeger, we alter this CR so this tracing.enabled is false. We can do this now since this PR is introducing our own copy of the CR on the filesystem (this replaces line 533):

```
if [ "${REMOVE_JAEGER} == "true" ]; then
   TRACING_ENABLED="false"
fi
cat ${MAISTRA_INSTALL_YAML} | TRACING_ENABLED={$TRACING_ENABLED:-true} envsubst | ${MAISTRA_ISTIO_OC_COMMAND} create -n istio-system -f -
```

Then change the line below (line 69 that says "enabled: true") to be `enabled: $TRACING_ENABLED`.

This --no-jaeger option is used by people (I know Micke uses it, I use it too) because Jaeger is resource-intensive and unless you are developing a feature dealing with tracing, its easier on our machines when we completely disable it.

Reply:
testing.. I see this as the output with --no-jaeger... we should put grafana at the end of the list. Also we should add "-n" option to the "Waiting for <pod name>" so the output is on one line

```
Wait for Istio to fully start (this is going to take a while)...Waiting for grafana
................................................done.
Waiting for istio-citadel
done.
Waiting for istio-egressgateway
done.
Waiting for istio-galley
done.
Waiting for istio-ingressgateway
done.
Waiting for istio-pilot
done.
Waiting for istio-policy
done.
Waiting for istio-sidecar-injector
done.
Waiting for istio-telemetry
done.
Waiting for prometheus
done.
Waiting for Istio Deployments to be created...Waiting for grafana to be ready....done.
Waiting for istio-citadel to be ready....done.
Waiting for istio-egressgateway to be ready....done.
Waiting for istio-galley to be ready....done.
Waiting for istio-ingressgateway to be ready....done.
Waiting for istio-pilot to be ready....done.
Waiting for istio-policy to be ready....done.
Waiting for istio-sidecar-injector to be ready....done.
Waiting for istio-telemetry to be ready....done.
Waiting for prometheus to be ready....done.

```